src/plonk/circuit.rs

```diff
@@ -283,21 +387,75 @@ impl<F: Field> ConstraintSystem<F> {
 
         // Return existing query, if it exists
         for (index, aux_query) in self.aux_queries.iter().enumerate() {
-            if aux_query == &(wire, at) {
+            if aux_query == &(column, at) {
                 return index;
             }
         }
 
         // Make a new query
         let index = self.aux_queries.len();
-        self.aux_queries.push((wire, at));
+        self.aux_queries.push((column, at));
 
         index
     }
 
-    /// Query an auxiliary wire at a relative position
-    pub fn query_aux(&mut self, wire: AuxWire, at: i32) -> Expression<F> {
-        Expression::Aux(self.query_aux_index(wire, at))
+    /// Query an Any column at a relative position
+    fn query_any_index(&mut self, column: Column<Any>, at: i32) -> usize {
+        let index = match column.column_type {
+            Any::Advice => self.query_advice_index(Column::<Advice>::from(column), at),
+            Any::Fixed => self.query_fixed_index(Column::<Fixed>::from(column), at),
+            Any::Aux => self.query_aux_index(Column::<Aux>::from(column), at),
+        };
+
+        index
+    }
+
+    /// Query an auxiliary column at a relative position
```

Comment:
For consistency, put this immediately below `query_aux_index` and add `query_any` here.